### DISCUSSION TITLE
Require Renovate version in config presets

### DISCUSSION THREAD

**Discussion by mbrodala** [renovatebot/renovate#26831] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.134.0

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### What is your question?

Is it possible to require a minimum version of Renovate in config presets?

Specifically I am asking because we are using a `regex` manager for GitLab CI/CD components which would be obsolete with version [37.148.0](https://github.com/renovatebot/renovate/releases/tag/37.148.0) of Renovate.

So we could drop this from our organization config but first need to make sure that Renovate is updated. This is not a real issue in this case since we are in control of the infrastructure.

However I wonder if there should be an option for config presets to require a minimum Renovate version exactly for cases like these. If Renovate then processes a repository which in the end requires a newer Renovate version, it should stop with an error.

It could look like this:

```json
{
    "renovate": {
        "minimumVersion": "37.148.0"
    }
}
```

(Opted for a NS-like `renovate` object here since there may be other options aiming at Renovate itself in the future.)

**Comment by rarkins** [comment]
Do you mean this?
- You have a custom regex manager which will no longer be needed
- Your regex manager is within a preset (dedicated? or a big shared one?)
- You want the regex manager (or preset?) ignored once Renovate is at or above a certain version?

**Reply by mbrodala** [reply]
Actually I want to drop the `regex` manager and require the mentioned Renovate version instead. This way one can continue using the config preset without worrying about missing updates.

If I wasn't in full control over the infrastructure, I could not drop the `regex` manager since I couldn't be sure about the used Renovate version. I'd instead need to wait a while :tm: before doing this and have :crossed_fingers: when dropping it eventually.

**Reply by rarkins** [reply]
Can you clarify/correct the three dot points?

**Reply by mbrodala** [reply]
Sure:

- You have a custom regex manager which will no longer be needed: yes
- Your regex manager is within a preset (dedicated? or a big shared one?): yes, a shared organization preset, [see here](https://github.com/pagemachine/renovate-config/blob/26dd5fa76604a269f8621d6cdf965518853a6e00/default.json#L13-L26)
- You want the regex manager (or preset?) ignored once Renovate is at or above a certain version?: no

**Reply by rarkins** [reply]
How does requiring a minimum Renovate version help your particular use case? I thought you'd want Renovate to _ignore_ that regex manager once it reaches a certain version. i.e. it sounded like you need to opt-out when you hit a minimum version, and not opt-in

**Reply by mbrodala** [reply]
I don't want to have the `regex` manager ignored, I want to completely drop it myself since I want it gone from our config preset. It's obsolete with the mentioned Renovate version after all, so keeping it around would be pointless.
